### PR TITLE
RandomSmith - Changes to allow user defined pins to be supplied

### PR DESCRIPTION
**Pull request by randomsmith9**
Hi,

I was trying to use the CE-PiicoDev-Unified and CE-PiicoDev-VL53L1X-MicroPython-Module libraries
but wanted to override the pins, which I wasn't sure If I could do due to the global variables for i2c.

I am fairly new to Python and MicroPython and microcontrollers, but developer by trade.
Feel free to use the code in the pull request if you like. It will provide separation of the code between different flavours of micropython and or boards.
It has an implementation for I2CUnifiedMicroBit which only includes code that is relevant to the micro:bit
and an implementation for IC2 from the machine library which should suit other boards. If it doesn't then it is easy to add extra classes like for the microbit or machine and adding extra classes means that you would be less likely to introduce errors because you will not likely be changing the microbit code when modifying or adding new boards - hope that makes sense.

I have also added some code in the readme.md file that shows how you would add the unified lib to your sensor so that users can use the defaults, but those users that need to change the pins can pass in pin details too...: i.e. 

from PiicoDev_Unified import create_unified_i2c
class PiicoDev_VL53L1X:

    def __init__(self, address=0x29, i2c=None):
        if i2c is not None:
            print("Using supplied i2c")
            self.i2c = i2c
        else:
            self.i2c = create_unified_i2c()

Saw the youtube video where Michael was asking about if the unified style code was a good idea
this will allow you extend more easily when new edge cases arise - https://youtu.be/2oinDOsrIO8?t=488

### CODE REVIEW

**Comment by randomsmith9** [comment]
TL;DR, the method names on your unified interface should match those of machine.I2C. since that would be the most prevalent interface.
Reasoning:
The other comment I meant to add is that for the unified interface, you are better using the most prevalent method names, i.e. likely the names on the machine.i2c so write should really be named writeto and read be called readfrom.
This means that any user who does not pass a unified I2C instance will still work because of duck typing. (i.e. the most likely object they are likely to pass that were not the unified I2C being machine.I2C which is the same interface so would just work)

**Comment by CoreProduction** [comment]
Hi! Thanks for opening the PR! I'm thrilled you want to contribute to our open source project.
FYI I've just merged a major update into the main branch that you may be interested in checking out. From what I gather we may be moving in a similar direction - though I'll need some time to digest your PR (Nice styling too 👍 )
I can see how your `init` will allow the user to select the i2c bus. I'll be interested to see how it rolls into the existing structure.

**Comment by CoreProduction2** [comment]
This has been addressed here https://github.com/CoreElectronics/CE-PiicoDev-Unified/pull/3
